ui/app/store/actions.js

```diff
@@ -1860,6 +1861,30 @@ function createSpeedUpTransaction (txId, customGasPrice) {
   }
 }
 
+function createRetryTransaction (txId, customGasPrice) {
+  log.debug('background.createRetryTransaction')
+  let newTx
+
+  return dispatch => {
+    return new Promise((resolve, reject) => {
+      // I'm kinda just piggybacking on `createSpeedUpTransaction` to avoid creating a new background method
+      // Not sure if this is acceptable?
```

Comment:
I think it is _acceptable_ and is definitely what I would were I assigned this task. A slight improvement could be changing the name to reflect a more general use case... I mean, `createRetryTransaction` does encompass speed ups...

But then again, we'd have to make it clear that if no custom gas price is provided, there is a default speed up applied. Perhaps the transaction controller method could be generalized by passing it a flag that indicates whether the default speed up should be applied? Then the inclusion of that flag in the method call would be somewhat self documenting...

However, as it stands, I think what is here is fine.